galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/CPSRoute.java

```diff
@@ -69,13 +70,22 @@ protected boolean isHiddenNamespace(String namespace){
         return hiddenNamespaces.contains(namespace);
     }
 
-    protected boolean isWriteOnlyNamespace(String namespace){
-        return writeOnlyNamespaces.contains(namespace);
+    protected String getNamespaceType(String namespace){
+        String type = NamespaceType.NORMAL.toString();
+        if (CPSRoute.isSecureNamespace(namespace)){
+            type= NamespaceType.SECURE.toString();
+        }
+        return type;
+    }
```

Comment:
If you'd like, you could return a NamespaceType here, and turn the Namespace class's `type` field into a NamespaceType instead of a string